### PR TITLE
LLD My Ledger: proper states for custom lock screen + rework file structure/naming & break down big components

### PR DESCRIPTION
**Pull request by ofreyssinet-ledger**
### 📝 Description

2 things are in this PR:

#### Reworked the file structure for LLD My Ledger
A small refactoring of the file structure of the My Ledger screen on LLD, because it was very outdated and all over the place.

_Before_
before

_After_
Untitled copy

#### Custom lock screen states (the initial point of the ticket I was tackling)

- handling the custom lock screen states properly in the manager
  - when a custom lock screen is set on the device, the "add" and "remove" buttons are replaced by a single "change" button
  - the "remove" feature is now accessible directly in the custom lock screen modal, after clicking change
  - the "change lock screen" button is disabled when apps are installing/uninstalling, and a tooltip explains the reason why the buttons are disabled. (also added this to the language and device renaming button, following the design)

https://github.com/LedgerHQ/ledger-live/assets/91890529/767d193d-839e-4340-a893-8af444e1bbfc

### ❓ Context

- **JIRA or GitHub link**: [LIVE-10339]

### ✅ Checklist

Pull Requests must pass the CI and be code reviewed. Set as Draft if the PR is not ready.

- [x] `npx changeset` was attached.
- [x] **Covered by automatic tests.** 
- [ ] **Impact of the changes:** 
  - 

---

### 🧐 Checklist for the PR Reviewers

- [x] **The code aligns with the requirements** described in the linked JIRA or GitHub issue.
- [x] **The PR description clearly documents the changes** made and explains any technical trade-offs or design decisions.
- [x] **There are no undocumented trade-offs**, technical debt, or maintainability issues.
- [ ] **The PR has been tested** thoroughly, and any potential edge cases have been considered and handled.
- [ ] **Any new dependencies** have been justified and documented.
- [ ] **Performance** considerations have been taken into account. (changes have been profiled or benchmarked if necessary)

[LIVE-10339]: https://ledgerhq.atlassian.net/browse/LIVE-10339?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/ledger-live-desktop/src/renderer/screens/manager/DeviceStorage/index.tsx
Comment (ofreyssinet-ledger) [inline apps/ledger-live-desktop/src/renderer/screens/manager/DeviceStorage/index.tsx:1]: this file was way too big and was what was rendering the full box containing device firmware info (cf picture), storage info, language, custom lock screen.
It became `DeviceInformationSummary/index.ts`, the components `StorageBar` and `StorageInfo` have been extracted in their own files.

Screenshot 2023-12-13 at 13 08 46

File: apps/ledger-live-desktop/src/renderer/screens/manager/DeviceDashboard/DeviceInformationSummary/StorageBar.tsx
Comment (ofreyssinet-ledger) [inline apps/ledger-live-desktop/src/renderer/screens/manager/DeviceDashboard/DeviceInformationSummary/StorageBar.tsx:1]: component extracted from the deleted file `DeviceStorage/index.tsx`

File: apps/ledger-live-desktop/src/renderer/screens/manager/DeviceDashboard/DeviceInformationSummary/StorageInfo.tsx
Comment (ofreyssinet-ledger) [inline apps/ledger-live-desktop/src/renderer/screens/manager/DeviceDashboard/DeviceInformationSummary/StorageInfo.tsx:1]: component extracted from the deleted file `DeviceStorage/index.tsx`

File: apps/ledger-live-desktop/src/renderer/screens/manager/DeviceDashboard/DeviceInformationSummary/index.tsx
Comment (ofreyssinet-ledger) [inline apps/ledger-live-desktop/src/renderer/screens/manager/DeviceDashboard/DeviceInformationSummary/index.tsx:1]: component rendering this

Screenshot 2023-12-13 at 13 08 46

code coming from the deleted file `DeviceStorage/index.ts`

File: apps/ledger-live-desktop/src/renderer/screens/manager/DeviceDashboard/DeviceInformationSummary/DeviceIllustration.tsx
Comment (ofreyssinet-ledger) [inline apps/ledger-live-desktop/src/renderer/screens/manager/DeviceDashboard/DeviceInformationSummary/DeviceIllustration.tsx:1]: component extracted from the deleted file DeviceStorage/index.tsx

**Review by alexandremgo — APPROVED** [review]
Great refactoring 🔥

**Review by mcayuelas-ledger — APPROVED** [review]
Nice :)

**Pull request merged**